packages/form-core/src/FormApi.ts

```diff
@@ -340,17 +337,16 @@ export class FormApi<
       void (
         Object.values(this.fieldInfo) as FieldInfo<any, TFormValidator>[]
       ).forEach((field) => {
-        Object.values(field.instances).forEach((instance) => {
-          // Validate the field
-          fieldValidationPromises.push(
-            Promise.resolve().then(() => instance.validate(cause)),
-          )
-          // If any fields are not touched
-          if (!instance.state.meta.isTouched) {
-            // Mark them as touched
-            instance.setMeta((prev) => ({ ...prev, isTouched: true }))
-          }
-        })
+        if (!field.instance) return
+        // Validate the field
+        fieldValidationPromises.push(
+          Promise.resolve().then(() => field.instance!.validate(cause)),
+        )
```

Comment:
We can get around the non-null assertion in `field.instance!` like this:

```suggestion
        if (!field.instance) return
        const fieldInstance = field.instance
        // Validate the field
        fieldValidationPromises.push(
          Promise.resolve().then(() => fieldInstance.validate(cause)),
        )
```